pdns/dnsname.cc

```diff
@@ -491,3 +491,61 @@ bool DNSName::has8bitBytes() const
 
   return false;
 }
+
+DNSName::RawLabelsVisitor::RawLabelsVisitor(const DNSName::string_t& storage): d_storage(storage)
+{
+  size_t position = 0;
+  while (position < storage.size()) {
+    auto labelLength = static_cast<unsigned char>(storage.at(position));
```

Comment:
For consistency (and pedantic correctness, as char might be multiple bytes):
```suggestion
    auto labelLength = static_cast<uint8_t>(storage.at(position));
```

Reply:
I sort of tripped on this one too. `std::string` is a `std::basic_string<char>` so `unsigned char` is more correct. I'm not sure about the boost variant.

Reply:
IMO the target (a position) is more relevant than the source value. The array holding positions is `std::array<uint8_t,128>`

Reply:
Right.. yes that makes sense

Reply:
Right, I'll make that change. Just for the record, though, POSIX mandates `char` to be exactly one byte :)